javascript/modules/client/src/client-allowlist.ts

```diff
@@ -0,0 +1,358 @@
+import { getAllowListProposalsWithStatus } from "./internal/utils/plugins";
+
+
+export class ClientAllowList extends ClientCore implements IClientAllowList {
+  private _pluginAddress: string;
+
+  constructor(context: ContextErc20) {
+    super(context);
+
+    if (!context.pluginAddress) {
+      throw new Error("An address for the plugin is required");
+    }
+    this._pluginAddress = context.pluginAddress;
+  }
+  methods = {
+    /**
+     * Creates a new proposal on the given AllowList plugin contract
+     *
+     * @param {ICreateProposalParams} _params
+     * @return {*}  {AsyncGenerator<ProposalCreationStepValue>}
+     * @memberof ClientAllowList
+     */
+    createProposal: (params: ICreateProposalParams): AsyncGenerator<ProposalCreationStepValue> =>
+      this._createProposal(params),
+    /**
+     * Cast a vote on the given proposal using the client's wallet. Depending on the proposal settings, an affirmative vote may execute the proposal's actions on the DAO.
+     *
+     * @param {string} proposalId
+     * @param {VoteOptions} vote
+     * @return {*}  {AsyncGenerator<VoteProposalStepValue>}
+     * @memberof ClientAllowList
+     */
+    voteProposal: (proposalId: string, vote: VoteOptions): AsyncGenerator<VoteProposalStepValue> =>
+      this._voteProposal(proposalId, vote),
+    /**
+     * Executes the given proposal, provided that it has already passed
+     *
+     * @param {string} proposalId
+     * @return {*}  {AsyncGenerator<ExecuteProposalStepValue>}
+     * @memberof ClientAllowList
+     */
+    executeProposal: (proposalId: string): AsyncGenerator<ExecuteProposalStepValue> =>
+      this._executeProposal(proposalId),
+    /**
+     * Sets the voting configuration in a allowlist proposal given a proposalId and a configuration
+     *
+     * @param {VotingConfig} config
+     * @return {*}  {AsyncGenerator<SetVotingConfigStepValue>}
+     * @memberof ClientAllowList
+     */
+    setPluginConfig: (config: VotingConfig): AsyncGenerator<SetVotingConfigStepValue> =>
+      this._setPluginConfig(config),
+    /**
+     * Returns the list of wallet addresses with signing capabilities on the plugin
+     *
+     * @return {*}  {Promise<string[]>}
+     * @memberof ClientAllowList
+     */
+    getMembers: (): Promise<string[]> =>
+      this._getMemebers(),
+    /**
+     * Returns the details of the given proposal
+     *
+     * @param {string} proposalId
+     * @return {*}  {Promise<AllowListProposal>}
+     * @memberof ClientAllowList
+     */
+    getProposal: (proposalId: string): Promise<AllowListProposal> =>
+      this._getProposal(proposalId),
+    /**
+     * Returns a list of proposals on the Plugin, filtered by the given criteria
+     *
+     * @param {IProposalQueryParams}
+     * @return {*}  {Promise<AllowListProposal[]>}
+     * @memberof ClientAllowList
+     */
+    getProposalMany: (params?: IProposalQueryParams): Promise<AllowListProposal[]> =>
+      this._getProposalMany(params ?? {}),
```

Comment:
At the risk of being inconsistent with myself in the past, wouldn't it look simpler to use `getProposals()` ? 

We have `getMetadataMany()` below, because there is no such a thing as "metadatas", but in such case, it would probably make sense to rename it to `getDao()` and `getDaos()`, metadata included